examples/basic_query_extractor/src/main.rs

```diff
@@ -0,0 +1,149 @@
+//! A basic example application for working with the Gotham Query String Extractor 
+
+extern crate futures;
+extern crate gotham;
+extern crate hyper;
+extern crate mime;
+extern crate serde;
+#[macro_use]
+extern crate serde_derive;
+extern crate serde_json;
+#[macro_use]
+extern crate gotham_derive;
+
+use hyper::{Response, StatusCode};
+
+use gotham::http::response::create_response;
+use gotham::router::Router;
+use gotham::router::builder::{build_simple_router, DefineSingleRoute, DrawRoutes};
+use gotham::state::State;
+
+/// `QueryParam` struct
+///
+/// It contains only a `name` field for the sake of simplicity
+/// It permits to use it as query parameter like:
+/// GET /widgets?name=t-shirt
+#[derive(Deserialize, StateData, StaticResponseExtender)]
+struct QueryParam {
+    name: String,
+}
+
+/// `Product` struct
+///
+/// It represents products that can be queried
+#[derive(Serialize)]
+struct Product {
+    name: String,
+    price: f32,
```

Comment:
How about we switch `price` for `description` for now? 

I wouldn't want to give the wrong impression about types that are considered useful in representing currency.